src/test/common/extensionMessaging.test.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {HostPlatform} from "../../common/hostPlatform";
+
+import {
+    ExtensionMessage,
+    ExtensionMessageSender
+} from "../../common/extensionMessaging";
+
+import * as assert from "assert";
+import * as net from "net";
+import * as sinon from "sinon";
+import * as Q from "q";
+
+suite("extensionMessaging", function() {
+    suite("commonContext", function() {
+        teardown(function() {
+            if (HostPlatform.getExtensionPipePath.hasOwnProperty("restore")) {
+                (<any>HostPlatform.getExtensionPipePath).restore();
+            }
+        });
+
+        test("should successfully send a message", function(done: MochaDone) {
+            let port: string = HostPlatform.getExtensionPipePath();
+            let mockServer: net.Server = net.createServer(function(client: net.Socket): void {
+                mockServer.close();
+                client.on("data", function(data: Buffer) {
+                    let messageData: any = JSON.parse(data.toString("utf8"));
+                    client.end();
+
+                    assert.equal(messageData.message, ExtensionMessage.START_PACKAGER);
+                });
+            });
+
+            mockServer.on("error", done);
+
+            mockServer.listen(port);
+
+            sinon.stub(HostPlatform, "getExtensionPipePath", function() {
```

Comment:
Since we now still use the default `getExtensionPipePath` value for this, is the stub still needed?